pydra/engine/state.py

```diff
@@ -77,7 +164,15 @@ class State:
 
     """
 
-    def __init__(self, name, splitter=None, combiner=None, other_states=None):
+    def __init__(
+        self,
+        name,
+        definition: "TaskDef",
```

Comment:
`State` used to be independent on the task. Not sure if I understand why you decided to add the task here. 
I could see that you're using it in one place instead of passing input, but wondering if it's really worth it

Reply:
You are right, I could probably just pass it into the method where it is used

Reply:
i believe you don't even need to pass it there, you just need input variables